[Doc] Update profiler (#6214)
taichi-dev/taichi: docs/lang/articles/performance_tuning/profiler.md

## Overview

> To facilitate code debugging and code optimization, Taichi provides a set of profiling tools, which collect hardware and Taichi-related information to quantify program performance and analyze where a bottleneck occurs.

Reviewer: ```suggestion
To facilitate code debugging and code optimization, Taichi provides a set of profiling tools, which collect hardware and Taichi-related information to quantify program performance and analyze where the bottleneck is.
```

Currently, Taichi provides two profiling tools:

- `ScopedProfiler` analyzes the performance of the Taichi JIT compiler (host).

> - `KernelProfiler` analyzes the performance of Taichi kernels (device). Its advanced mode, which is applicable to the CUDA backend only, provides detailed low-level performance metrics, such as memory bandwidth consumption.

Reviewer: "apply to" is ALWAYS more succinct than "is applicable to"
```suggestion
- `KernelProfiler` analyzes the performance of Taichi kernels (device). Its advanced mode, which works with the CUDA backend only, provides detailed low-level performance metrics, such as memory bandwidth consumption.
```

:::note

- `jit_evaluator_xxx` can be ignored because it is automatically generated by the system.

> - You are recommended running performance profiling multiple times to observe the minimum or average execution time.

Reviewer: I edited this line before. 
Correct: "Taichi recommends doing" or "Taichi recommends that you do". 
Incorrect: You are recommended doing. 

At least it ought to be "you are recommended to do".

Reviewer: I suggest you to download a plugin called GitLens from vs code marketplace to avoid duplicate edits.
